### PR TITLE
P2PW-1571 - Add caching on Transactions History screen

### PR DESCRIPTION
**Pull request by gslevinkov**
the original task was reviewed before, this one needs to be tested and merged into dev

### CODE REVIEW

**Review by kabramovp2p — APPROVED** [review]

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/history/interactor/TransactionsHistoryRepository.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+        return transactionsFromRemote
+    }
+
+    private fun getAccountsInfo(
+        fetchedTransactions: List<TransactionDetails>
+    ): List<Pair<String, AccountInfo>> {
+        // Making one request for all accounts info and caching values locally
+        // to avoid multiple requests when constructing transaction
+        val accountsInfoIds = fetchedTransactions
```
Comment (davran1001) [inline app/src/main/java/org/p2p/wallet/history/interactor/TransactionsHistoryRepository.kt:68]: тут можно asSequence заюзать

**Review by davran1001 — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/history/interactor/TransactionsHistoryRepository.kt (hunk 1 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/history/interactor/TransactionsHistoryRepository.kt]: i don't think that would optimize things

File: app/src/main/java/org/p2p/wallet/history/repository/HistoryTransactionConverter.kt [hunk 2]
```diff
@@ -67,19 +66,23 @@ object TransactionConverter {
             destinationIconUrl = destinationData.iconUrl.orEmpty()
         )
 
+    private fun String.toBigDecimalFromLamports(decimals: Int): BigDecimal =
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/history/repository/HistoryTransactionConverter.kt:69]: Did you mean `toBigIntegerFromLamports`?

File: app/src/main/java/org/p2p/wallet/history/repository/HistoryTransactionConverter.kt [hunk 3]
```diff
@@ -18,10 +21,10 @@ import org.threeten.bp.ZoneId
 import org.threeten.bp.ZonedDateTime
 import java.math.BigDecimal
 
-object TransactionConverter {
+class HistoryTransactionConverter {
 
     /* Swap transaction */
-    fun fromNetwork(
+    fun mapSwapTransactionToHistory(
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/history/repository/HistoryTransactionConverter.kt:27]: Did we decide to move from `fromNetwork`/`toNetwork` style?

File: app/src/main/java/org/p2p/wallet/utils/SolanajTimberLogger.kt [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package org.p2p.wallet.utils
+
+import org.p2p.solanaj.utils.SolanjLogger
+import timber.log.Timber
+
+private const val SOLANAJ_TAG = "Solanaj module"
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/utils/SolanajTimberLogger.kt:6]:
```suggestion
private const val SOLANAJ_TAG = "SOLANAJ_MODULE"
```
I think this format will easier to find in logcat

**Review by 1sakov — CHANGES_REQUESTED** [review]

**Comment by 1sakov** [comment]
@gslevinkov added minor comments

File: app/src/main/java/org/p2p/wallet/history/repository/HistoryTransactionConverter.kt (hunk 2 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/history/repository/HistoryTransactionConverter.kt:69]: return type is BigDecimal, so i think it's alright here

File: app/src/main/java/org/p2p/wallet/history/repository/HistoryTransactionConverter.kt (hunk 3 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/history/repository/HistoryTransactionConverter.kt:27]: it's not about network here, but about mapping domain entity to another domain entity

File: app/src/main/java/org/p2p/wallet/history/repository/HistoryTransactionConverter.kt (hunk 2 above)
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/history/repository/HistoryTransactionConverter.kt:69]: Oh, yeah, my fault

**Review by 1sakov — APPROVED** [review]

**Pull request merged**